src/test/java/com/imagination/cbs/service/impl/EmailServiceImplTest.java

```diff
@@ -0,0 +1,335 @@
+		
+		emailServiceImpl.sendInternalResourceEmail(internalResourceEmailDto);
+
+		verify(config).getTemplate("email.internalsource.ftl");
+	}
+	
+	@Test
+	public void shouldLogExceptionMessageOccuredDuringEmailSending_sendInternalResourceEmail() {
+		
+		InternalResourceEmailDto internalResourceEmailDto = createInternalResourceEmailDto();
+		
+		emailServiceImpl.sendInternalResourceEmail(internalResourceEmailDto);
+	}
+	
+	private MailRequest createMailRequest() {
+
+		MailRequest emailRequest = new MailRequest();
+		emailRequest.setMailFrom("CBS@imagination.com");
+		emailRequest.setMailTo(new String[] {"CBS_Receiver@imagination.com"});
+		emailRequest.setSubject("Test mail");
+		
+		return emailRequest;
+	}
+	
+	private BookingRevision createBookingRevision(boolean isEmptyContent) {
+		
+		BookingRevision bookingRevision = new BookingRevision();
+		bookingRevision.setContractAmountAftertax(new BigDecimal("1000")); 
+		
+		ContractorEmployee contractEmployee = new ContractorEmployee(); 
+		if(!isEmptyContent) {
+			contractEmployee.setContractorEmployeeName("Akshay"); 
+		}
+		bookingRevision.setContractEmployee(contractEmployee);
+
+		Contractor contractor = new Contractor(); 
+		if(!isEmptyContent) {
+			contractor.setContractorName("Imagination"); 
+		}
+		bookingRevision.setContractor(contractor);
+
+		RoleDm roleDm = new RoleDm();
+		roleDm.setRoleName("2D");
+		Discipline discipline = new Discipline();
+		discipline.setDisciplineName("Creative");
+		roleDm.setDiscipline(discipline);
+		bookingRevision.setRole(roleDm);
+		
+		SupplierTypeDm supplierType = new SupplierTypeDm();
+		if(!isEmptyContent) {
+			supplierType.setName("Yash"); 
+		}
+		bookingRevision.setSupplierType(supplierType);
+		
+		Timestamp timeStamp = new Timestamp(System.currentTimeMillis());
+		bookingRevision.setContractedFromDate(timeStamp);
+		bookingRevision.setContractedToDate(timeStamp);
+		
+		OfficeDm officeDm = new OfficeDm();
+		if(!isEmptyContent) {
+			officeDm.setOfficeName("India"); 
+		}
+		bookingRevision.setContractWorkLocation(officeDm);
+		
+		ReasonsForRecruiting reasonForRecruiting = new ReasonsForRecruiting();
+		if(!isEmptyContent) {
+			reasonForRecruiting.setReasonName("Specific skills Required"); 
+		}
+		bookingRevision.setReasonForRecruiting(reasonForRecruiting);
+		
+		if(!isEmptyContent) {
+			List<BookingWorkTask> bookingWorkTasks = getBookingWorkTasks();
+			bookingRevision.setBookingWorkTasks(bookingWorkTasks);
+		}
+		
+		Booking booking = new Booking();
+		booking.setBookingId(101L);
+		booking.setChangedBy("Akshay");
+		bookingRevision.setBooking(booking);
+		
+		ApprovalStatusDm approvalStatusDm = new ApprovalStatusDm();
+		approvalStatusDm.setApprovalStatusId(201L);
+		approvalStatusDm.setApprovalName("Mitul");
+		bookingRevision.setApprovalStatus(approvalStatusDm);
+
+		return bookingRevision;
+	}
+	
+	private BookingRevision createBookingRevisionWithNullObjects() {
```

Comment:
these methods are way bigger, is there any possibility to break this in small methods.

Reply:
These are creating Dtos according the required inputs

Reply:
alright ... thats why i said if we can move the redundant code like 
RoleDm roleDm = new RoleDm();
		roleDm.setRoleName("2D");
		Discipline discipline = new Discipline();
		discipline.setDisciplineName("Creative");
		roleDm.setDiscipline(discipline);
		bookingRevision.setRole(roleDm);

to some common place in a static method and then can call it anywhere in our test, this will make code short and less redundant. what do you think ?

Reply:
Did changes according to suggestion